carbon-mss/components/org.wso2.carbon.mss/src/main/java/org/wso2/carbon/mss/internal/router/HttpResourceModelProcessor.java

```diff
@@ -0,0 +1,198 @@
+                                              String contentType,
+                                              List<String> acceptTypes)
+            throws HandlerException {
+
+        //TODO: Refactor group values.
+        try {
+            if (httpResourceModel.getHttpMethod().contains(request.getMethod())) {
+                //Setup args for reflection call
+                List<HttpResourceModel.ParameterInfo<?>> paramInfoList = httpResourceModel.getParamInfoList();
+                List<String> producesMediaTypes = httpResourceModel.getProducesMediaTypes();
+                Object[] args = new Object[paramInfoList.size()];
+                String acceptType = "*/*";
+                if (!producesMediaTypes.contains("*/*") && acceptTypes != null) {
+                    acceptType =
+                            (acceptTypes.contains("*/*")) ? producesMediaTypes.get(0) :
+                                    producesMediaTypes.stream().filter(acceptTypes::contains).findFirst().get();
+                }
+                int idx = 0;
+                for (HttpResourceModel.ParameterInfo<?> paramInfo : paramInfoList) {
+                    if (paramInfo.getAnnotation() != null) {
+                        Class<? extends Annotation> annotationType = paramInfo.getAnnotation().annotationType();
+                        if (PathParam.class.isAssignableFrom(annotationType)) {
+                            args[idx] = getPathParamValue((HttpResourceModel.ParameterInfo<String>) paramInfo,
+                                    groupValues);
+                        } else if (QueryParam.class.isAssignableFrom(annotationType)) {
+                            args[idx] = getQueryParamValue((HttpResourceModel.ParameterInfo<List<String>>) paramInfo,
+                                    request.getUri());
+                        } else if (HeaderParam.class.isAssignableFrom(annotationType)) {
+                            args[idx] = getHeaderParamValue((HttpResourceModel.ParameterInfo<List<String>>) paramInfo,
+                                    request);
+                        } else if (Context.class.isAssignableFrom(annotationType)) {
+                            args[idx] = getContextParamValue((HttpResourceModel.ParameterInfo<Object>) paramInfo,
+                                    request, responder);
+                        }
+                    } else if (request instanceof FullHttpRequest) {
+                        // If an annotation is not present the parameter is considered a
+                        // request body data parameter
+                        String content = ((FullHttpRequest) request).content().toString(Charsets.UTF_8);
+                        Type paramType = paramInfo.getParameterType();
+                        args[idx] = BeanConverter.instance((contentType != null) ? contentType : "*/*")
+                                .toObject(content, paramType);
```

Comment:
It is better to have a String constant for "*/*" since you have used it in many places again and again..